Jenkinsfile

```diff
@@ -1,89 +1,162 @@
 #!groovy
-library "atomSharedLibraries@master"
+library "atomSharedLibraries@${env.BRANCH_NAME}"
 
-import static Constants.*
-
-class Constants {
-    static final mainDirectoryName = "contrail_state_files"
-    static final terraformStateFileName = "state"
-}
+@Library("atomSharedLibraries@${env.BRANCH_NAME}")
+import org.FileManager
+import org.DirKeeper
+import org.FileNameKeeper
+import org.Constants
 
 def warningEcho(message) {
     echo "\033[1;33m${message}\033[0m"
 }
 
-def ensureMainDirectoryExists() {
-    dirExistingResult = checkIfDirectoryExists("../${mainDirectoryName}")
-    if(dirExistingResult == '0') {
-        sh "mkdir ../${mainDirectoryName}"
-    }
+def copyTerraformFiles(FileManager fm, DirKeeper dk, FileNameKeeper fnk) {
+    fm.copy(fnk.tfMainName(), dk.tfFilesDir(), dk.machineDir())
+    fm.copy(fnk.tfVarsName(), dk.tfFilesDir(), dk.machineDir())
 }
 
-def ensureUserDirectoryExists() {
-    dirExistingResult = checkIfDirectoryExists("../${mainDirectoryName}/${params.Login}")
-    if(dirExistingResult == '0') {
-        sh "mkdir ../${mainDirectoryName}/${params.Login}"
-    }
+def copyDaemonFile(FileManager fm, DirKeeper dk, FileNameKeeper fnk) {
+    fm.copy(fnk.daemonName(), dk.provisionDir(), dk.machineDir())
 }
 
-def checkIfDirectoryExists(name) {
-    return sh(script: "test -d \"${name}\" && echo '1' || echo '0' ", returnStdout: true).trim()
+def resolveInstancesYamlFile(FileManager fm, DirKeeper dk, FileNameKeeper fnk) {
+    if (fnk.userInstanceName() != "") {
+        fm.move(fnk.userInstanceName(), "${dk.machineDir()}/${fnk.mainInstanceName()}")
+    } else {
+        warningEcho("Didn't get instances yaml file. Using default file.")
+        fm.copy("${dk.provisionDir()}/${fnk.defaultInstanceName()}", "${dk.machineDir()}/${fnk.mainInstanceName()}")
+    }
 }
 
-def checkIfFileExists(name) {
-    return sh(script: "test -f \"${name}\" && echo '1' || echo '0' ", returnStdout: true).trim()
+def prepareMachineDirectory(FileManager fm, DirKeeper dk, FileNameKeeper fnk) {
+    fm.newDir(dk.machineDir())
+    sh "chmod 500 ${dk.provisionDir()}/${fnk.prepareTemplateName()}"
+    fm.copy(fnk.prepareTemplateName(), dk.provisionDir(), dk.machineDir())
+    copyTerraformFiles(fm, dk, fnk)
+    copyDaemonFile(fm, dk, fnk)
+    resolveInstancesYamlFile(fm, dk, fnk)
+    prepareKeyFiles(fm, dk, fnk)
 }
 
-def copyTerraformFiles() {
-    sh "cp provision/${params.orchestrator}/${params.orchestrator}.tf ../${mainDirectoryName}/${params.Login}/${params.MachineName}/${params.orchestrator}.tf"
-    sh "cp provision/${params.orchestrator}/variables.tf ../${mainDirectoryName}/${params.Login}/${params.MachineName}/variables.tf"
+def prepareKeyFiles(FileManager fm, DirKeeper dk, FileNameKeeper fnk) {
```

Comment:
What we have here is that we have split functionality 'horizontally' - we have a class for handlig files, dirs and names.

But look at this function, it takes all of these classes. Wouldn't it be better to have a class like, I dunno `SecretsManager`, which has `prepareKeyFiles` method? 

Splitting functionality this way would create actual classes, not structs. Class would then encompass all the fields required for only its responsibility. E.g. `SecretsManager` would have all the paths for keys or whatever stored inside, and some methods that operate on them, and that's it.

This thinking could be applied to the whole diff.